content/en/docs/tasks/administer-cluster/migrating-from-dockershim/ migrating-telemetry-and-security-agents.md

```diff
@@ -0,0 +1,99 @@
+
+Docker is being deprecated. From the [Dockershim Deprecation
+FAQ](https://kubernetes.io/blog/2020/12/02/dockershim-faq/) you might already
+know that most apps do not have a direct dependency on runtime hosting
+containers. However, there are still a lot of telemetry and security agents
+that has a dependency on docker to collect containers metadata, logs and
+metrics. This document aggregates information on how to detect these
+dependencies and links on how to migrate these agents to use generic tools or
+alternative runtimes.
+
+## Telemetry and security agents
+
+There are a few ways agents may be run on kubernetes cluster. Agents have a direct dependency on Docker when they run on nodes or as daemonsets. Some telemetry and security agents may rely on certain logs or metrics specific for Docker that would not be available any longer. This document does not cover these cases.
+
+### Why do telemetry agents talk to Docker?
+
+Historically, kubernetes was built on top of Docker. Kubernetes is managing networking and scheduling, Docker was placing and operating containers on a node. So you can get scheduling-related metadata like a pod name from Kubernetes and containers state information from Docker. Over time more runtimes were created to manage containers. Also there are projects and kubernetes features that generalize container status information extraction across many runtimes.
+
+But some agents are still Docker-dependent. They may run commands like [docker
+ps](https://docs.docker.com/engine/reference/commandline/ps/) or [docker
+top](https://docs.docker.com/engine/reference/commandline/top/) to list
+containers and processes or [docker
+logs](https://docs.docker.com/engine/reference/commandline/logs/) to subscribe
+on docker logs. With the deprecating of Docker as a container runtime, these
+commands will not work any longer.
+
+
+### Detecting Docker dependency from the agents-daemonsets
+
+If a pod wants to make calls to the `dockerd` running on the node, the pod must mount the docker daemon as a volume.
+
+For example, On COS images, docker exposes it’s unix socket at
+`/var/run/docker.sock` This means that the customer’s pod spec will include a
+`hostPath` volume mount of `/var/run/docker.sock`.
+
+The script to find all pods mounting this volume may look like this:
+
+```bash
+kubectl get pods --all-namespaces \
+-o=jsonpath='{range .items[*]}{"\n"}{.metadata.namespace}{":\t"}{.metadata.name}{":\t"}{range .spec.volumes[*]}{.hostPath.path}{", "}{end}{end}' \
+| sort \
+| grep '/var/run/docker.sock'
+```
+
+(this script outputs the namespace and name of the pod. You can remove the grep `/var/run/docker.sock` to review other mounts)
+
+Note, there are alternative ways to call a Docker. For instance, the parent
+directory `/var/run` may be mounted instead of the full path (like in [this
+example](https://gist.github.com/itaysk/7bc3e56d69c4d72a549286d98fd557dd)). So
+the script above will only detect the most common uses.
```

Comment:
```suggestion
{{< note >}}
There are alternative ways for a pod to access Docker on the host. For instance, the parent
directory `/var/run` may be mounted instead of the full path (like in [this
example](https://gist.github.com/itaysk/7bc3e56d69c4d72a549286d98fd557dd)).
The script above only detects the most common uses.
{{< /note >}}
```

Is this acceptable for a task page? If we only explain some of the easier cases, readers might feel short changed.